### PR TITLE
Update the compiler to the 3.8.0 release

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -23,11 +23,12 @@
     <ReportGeneratorVersion>4.3.6</ReportGeneratorVersion>
     <!-- Use the correct compiler version -->
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
-    <CodeStyleAnalyersVersion>3.8.0-3.20428.4</CodeStyleAnalyersVersion>
+    <MicrosoftNetCompilersToolsetVersion>3.8.0</MicrosoftNetCompilersToolsetVersion>
```
Comment (Evangelink) [inline eng/Versions.props:26]: This seems to be a new property but there is no file updated to be using it.

Comment (mavasani) [reply eng/Versions.props:26]: This was explicitly removed by @mmitche as this causes a dependency chain issue when inserting analyzers into the SDK repo. We have been asked explicitly to not reference `MicrosoftNetCompilersToolsetVersion`, but instead use in-built compiler toolset from arcade version. Also adding @jmarolf

**Review by mavasani — CHANGES_REQUESTED** [review]
We have been asked to not reference `MicrosoftNetCompilersToolsetVersion`

File: eng/Versions.props (hunk 1 above)
Comment (sharwell) [reply eng/Versions.props:26]: The Microsoft.Net.Compilers.Toolset dependency does not propagate transitively.

Comment (mavasani) [reply eng/Versions.props:26]: @sharwell Let me add you to the email thread.

**Review by mavasani — APPROVED** [review]

**Comment by Evangelink** [comment]
@mavasani So is it ok to use `MicrosoftNetCompilersToolsetVersion`?

**Comment by mavasani** [comment]
@Evangelink Based on offline conversations, it was agreed it is fine. When we are nearing an official release as part of .NET SDK, we would have to remove it.

**Pull request merged**